explorer/interpreter/interpreter.cpp

```diff
@@ -1087,7 +1085,23 @@ auto Interpreter::StepExp() -> ErrorOr<Success> {
       if (auto rewrite = op.rewritten_form()) {
         return todo_.ReplaceWith(std::make_unique<ExpressionAction>(*rewrite));
       }
-      if (act.pos() != static_cast<int>(op.arguments().size())) {
+      if ((op.op() == Operator::And || op.op() == Operator::Or) &&
```

Comment:
Not sure this is the most elegant way to go about it, but tricky to fallthrough to default argument evaluation, as we have logic for both `pos() < size` and `pos() == size`.

Reply:
I think it might be clearer to enter this block only if `act.pos() == 1` and fall through if we don't short-circuit, and add back the support in `EvalPrim` for `And` and `Or`. (That is: if the first operand causes short-circuiting then do it here as a special case and finish evaluation, and otherwise treat `and` and `or` like any other binary operators.)

Reply:
Right, that's definitely simpler, was my first version of the implementation. Maybe a bit misleading in the sense that `EvalPrim` will look it is dealing with `and`/`or`, even though it is not, otherwise simpler if acceptable.
Any opinion @jonmeow , given this was your original intent?

Reply:
I pushed a version that goes in that direction. Pretty small changeset, I think it is worth it, and the performance impact of the superfluous `&&` + call is probably negligeable in the context of the explorer.